# Starbucks reviews ☕️☕️


1. Import pandas

In [1]:
import pandas as pd

2. Read the data from starbucks_reviews.csv

In [2]:
reviews_df = pd.read_csv('./starbucks_reviews.csv')
reviews_df.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review                date  
0  Amber and LaDonna at the Starbucks on Southwes...  September 13, 2023  
1  ** at the Starbucks by the fire station on 436...       July 16, 2023  
2  I just wanted to go out of my way to recognize...        July 5, 2023  
3  Me and my friend were at Starbucks and my card...        May 26, 2023  
4  I’m on this kick of drinking 5 cups of warm wa...    January 22, 2023

3. How many columns/lines are in the dataset?

In [3]:
num_rows, num_cols = reviews_df.shape
print(f"The dataset has {num_rows} rows and {num_cols} columns.")

The dataset has 850 rows and 5 columns.


4. Convert the column "date" into datetime type.

💡 You'll have to specify the format with python's [strptime syntax](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior)

In [33]:
import datetime
dt = datetime.datetime

reviews_df['date'] = reviews_df['date'].apply(dt.strptime, args=('%B %d, %Y',))
# type(reviews_df['date'][0])
# reviews_df.head()

pandas._libs.tslibs.timestamps.Timestamp

5. Extract some features from the "date" column:
* year
* month
* day of week

In [28]:
reviews_df.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  
0  Amber and LaDonna at the Starbucks on Southwes... 2023-09-13  
1  ** at the Starbucks by the fire station on 436... 2023-07-16  
2  I just wanted to go out of my way to recognize... 2023-07-05  
3  Me and my friend were at Starbucks and my card... 2023-05-26  
4  I’m on this kick of drinking 5 cups of warm wa... 2023-01-22

In [34]:
reviews_df['year'] = reviews_df['date'].apply(lambda a : a.year)
reviews_df['month'] = reviews_df['date'].apply(lambda a : a.month)
reviews_df['weekday'] = reviews_df['date'].apply(lambda a : a.day_of_week)
reviews_df.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  Amber and LaDonna at the Starbucks on Southwes... 2023-09-13  2023      9   
1  ** at the Starbucks by the fire station on 436... 2023-07-16  2023      7   
2  I just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  Me and my friend were at Starbucks and my card... 2023-05-26  2023      5   
4  I’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   weekday  
0        2  
1        6  
2        2  
3        4  
4        6

6. Convert the "Review" column into lowercase

In [37]:
reviews_df['Review'] = reviews_df['Review'].apply(lambda a : a.lower())
reviews_df.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  amber and ladonna at the starbucks on southwes... 2023-09-13  2023      9   
1  ** at the starbucks by the fire station on 436... 2023-07-16  2023      7   
2  i just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  me and my friend were at starbucks and my card... 2023-05-26  2023      5   
4  i’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   weekday  
0        2  
1        6  
2        2  
3        4  
4        6

7. Extract features from the "Review" column:
* Count the occurrences of the word "starbucks"
* Count the total number of words in each review

In [39]:
reviews_df['count_starbucks'] = reviews_df['Review'].apply(lambda a : a.count('starbucks'))
reviews_df['wordcount'] = reviews_df['Review'].apply(lambda a : len(a))

reviews_df.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  amber and ladonna at the starbucks on southwes... 2023-09-13  2023      9   
1  ** at the starbucks by the fire station on 436... 2023-07-16  2023      7   
2  i just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  me and my friend were at starbucks and my card... 2023-05-26  2023      5   
4  i’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   weekday  count_starbucks  wordcount  
0        2                1        331  
1        6                2        555  
2        2                1        351  
3        4                1        441  
4        6                3        405

8. Use pandas' `.describe()` method to get some basic statistics about the dataset and answer the questions below:
* Are the reviews rather good or bad?
* Are the reviews recent or old?
* Are the reviews rather long or short?
* Do people mention a lot the word "starbuck" ?

In [40]:
reviews_df.describe()

Rating                           date         year       month  \
count  705.000000                            850   850.000000  850.000000   
mean     1.870922  2016-02-17 12:38:57.882352896  2015.600000    6.897647   
min      1.000000            2000-07-19 00:00:00  2000.000000    1.000000   
25%      1.000000            2012-09-26 18:00:00  2012.000000    4.000000   
50%      1.000000            2016-05-31 00:00:00  2016.000000    7.000000   
75%      2.000000            2019-01-06 06:00:00  2019.000000   10.000000   
max      5.000000            2023-09-14 00:00:00  2023.000000   12.000000   
std      1.397672                            NaN     4.258436    3.400633   

          weekday  count_starbucks    wordcount  
count  850.000000       850.000000   850.000000  
mean     2.955294         1.241176   471.064706  
min      0.000000         0.000000    14.000000  
25%      1.000000         0.000000   272.000000  
50%      3.000000         1.000000   459.500000  
75%      5.000000         2.000000   655.250000  
max      6.000000         9.000000  1161.000000  
std      1.961437         1.248584   243.541473

### Are the reviews good or bad ?
The reviews seem to be rather bad, since the mean rating is 1.87 which is below the standard mean of 2.5 in the rating interval, and 75% of the values in the dataset are below 2
### Are the reviews recent or old ?
50 % Of the reviews are in the 7 years between 2012 and 2019 with only 25% for the 12 years between 2000 and 2012, and 25% for the 8 years between 2019 and 2023. This means that the vast majority of reviews are less than 11 years old, on a sample where the earliest reviews are 23 years old. I would consider that distribution not really recent, given the reviews of the last 4 years are only 25% of the dataset.
### Are the reviews long or short ?
Most reviews are between 273 and 655 words long, which given a minimum of 14 and the fact that "Great coffee" probably constitutes a valid review, tells me they are rather long.
